services/slam/fake/slam_test.go

```diff
@@ -0,0 +1,173 @@
+package fake
+
+import (
+	"context"
+	"testing"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+func TestFakeSLAMPosition(t *testing.T) {
+	slamSvc := &SLAM{Name: "test", logger: golog.NewTestLogger(t)}
+	pInFrame, err := slamSvc.Position(context.Background(), slamSvc.Name, map[string]interface{}{})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, pInFrame.Parent(), test.ShouldEqual, slamSvc.Name)
+
+	// These test ranges as I observed different floating point values
+	// between M1 mac hosts and amd64 hosts for the same test data.
+	test.That(t, pInFrame.Pose().Point().X, test.ShouldBeBetween, 1.860775180178, 1.860775180179)
```

Comment:
I hate testing this way but testing the exact floating point values yielded slightly different results between m1 macs & arm64 ci servers.